lib/Server.js

```diff
@@ -120,6 +119,7 @@ function Server(compiler, options) {
 
 		proxy: function() {
 			if (options.proxy) {
+				var proxy = new httpProxy.createProxyServer({secure: options.secureProxy});
 				if (!Array.isArray(options.proxy)) {
```

Comment:
http-proxy enables this security enforcement which is disabled by default in Node.js unless `secure: false` is passed.